src/activities/assignments/AssignmentEntity.js

```diff
@@ -201,6 +201,22 @@ export class AssignmentEntity extends Entity {
 		return subEntity.hasClass(Classes.assignments.assignmentType.readOnly);
 	}
 
+	/**
+	 * @returns {bool} If the assignment has submissions
+	 */
+	assignmentHasSubmissions() {
+		if (!this._entity) {
+			return false;
+		}
+
+		const subEntity = this._entity.getSubEntityByRel(Rels.Assignments.folderType);
+		if (!subEntity) {
+			return false;
```

Comment:
Aside: this is a case where `undefined` seems appropriate - the absence of a value! It might very well have submissions!

Don't actually change it though, just an observation.

Reply:
Hmm should we maybe actually default to true? Since if we don't know if an assignment has submissions we may attempt to change its properties when it actually could have submissions?

Reply:
The server is the _authority_ on whether you can take an action. If it provides an action, assume that you can do it. Otherwise, assume that you can't. Inferring this information through classes, etc. is what I'm raising in my top-level comment above.

In short: if action, can do, otherwise, no can do! Let's not rely on classes for things built-in to Siren.